commit c00c67bf29c10862a973289486a535c8566d5110
Author: Martin Gräßlin <mgraesslin@kde.org>
Date:   Wed Sep 3 09:47:21 2014 +0200

    [scripting] Drop qmlRegisterType for QQuickWindow
    
    Not needed as one can just do:
    import QtQuick.Window;
    
    This fixes a build problem with Qt 5.4.
    
    BUG: 338277

diff --git a/scripting/scripting.cpp b/scripting/scripting.cpp
index 3bc4358..9dc7bad 100644
--- a/scripting/scripting.cpp
+++ b/scripting/scripting.cpp
@@ -620,7 +620,6 @@ void KWin::Scripting::init()
     qmlRegisterType<KWin::ScriptingClientModel::ClientModelByScreenAndDesktop>("org.kde.kwin", 2, 0, "ClientModelByScreenAndDesktop");
     qmlRegisterType<KWin::ScriptingClientModel::ClientFilterModel>("org.kde.kwin", 2, 0, "ClientFilterModel");
     qmlRegisterType<KWin::Client>();
-    qmlRegisterType<QQuickWindow>();
     qmlRegisterType<QAbstractItemModel>();
 
     m_qmlEngine->rootContext()->setContextProperty(QStringLiteral("workspace"), m_workspaceWrapper);
diff --git a/scripts/desktopchangeosd/contents/ui/osd.qml b/scripts/desktopchangeosd/contents/ui/osd.qml
index fbf7dd7..e2ff86c 100644
--- a/scripts/desktopchangeosd/contents/ui/osd.qml
+++ b/scripts/desktopchangeosd/contents/ui/osd.qml
@@ -18,6 +18,7 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *********************************************************************/
 import QtQuick 2.0;
+import QtQuick.Window 2.0;
 import org.kde.plasma.core 2.0 as PlasmaCore;
 import org.kde.plasma.components 2.0 as Plasma;
 import org.kde.kquickcontrolsaddons 2.0 as KQuickControlsAddons;
